src/parser.rs

```diff
@@ -630,6 +636,12 @@ impl<'a> StatementRef<'a> {
         self.statement.stype
     }
 
+    /// Returns true if this statement is an `Axiom` (`$a`) or `Provable` (`$p`) statement.
+    #[must_use]
+    pub const fn is_assertion(self) -> bool {
+        self.statement.stype.is_assertion()
+    }
+
```

Comment:
This could be retrofitted in existing tests, e.g. `verify.rs` line 325, `proof.rs` line 538, and possibly others.